logicrunner/handle_calls.go

```diff
@@ -158,6 +157,30 @@ func (h *HandleCall) handleActual(
 	if objRef == nil {
 		return nil, errors.New("can't get object reference")
 	}
+	if !objRef.Record().Equal(reqInfo.ObjectID) {
+		return nil, errors.New("object id we calculated doesn't match ledger")
+	}
+
+	repl := &reply.RegisterRequest{Request: *requestRef}
+
+	if reqInfo.Request != nil && len(reqInfo.Request) != 0 {
+		logger.Debug("duplicated request")
+	}
+
+	if reqInfo.Result != nil && len(reqInfo.Result) != 0 {
+		logger.Debug("request has result already")
+		go func() {
+			err := h.sendRequestResult(ctx, *objRef, *requestRef, request, *reqInfo)
```

Comment:
Shouldn't we check that this was not a NoWait call?

Reply:
send will deal with this.